pkg/api/v1alpha1/tinkerbellmachineconfig.go

```diff
@@ -125,3 +127,27 @@ func validateTinkerbellMachineConfig(config *TinkerbellMachineConfig) error {
 
 	return nil
 }
+
+func setTinkerbellMachineConfigDefaults(machineConfig *TinkerbellMachineConfig) {
+	if len(machineConfig.Spec.Users) <= 0 {
+		machineConfig.Spec.Users = []UserConfiguration{{}}
+	}
+
+	if len(machineConfig.Spec.Users[0].SshAuthorizedKeys) <= 0 {
+		machineConfig.Spec.Users[0].SshAuthorizedKeys = []string{""}
```

Comment:
So we are setting the sshauthorized keys for the first element separately from the name value? Shouldn't we just set it at the same time? Also if this is default to empty, do we generate an ssh key?

Reply:
Currently if it's empty we don't generate keys for any providers via the controller. We are prioritizing that effort soon though to generate the keys if the user doesn't specify the key.